protocol/state/snapshot.go

```diff
@@ -13,6 +13,7 @@ type PriorIssuances map[bc.Hash]uint64
 // Snapshot encompasses a snapshot of entire blockchain state. It
 // consists of a patricia state tree and the issuances memory.
 type Snapshot struct {
+	B1Hash    bc.Hash
```

Comment:
No, sorry, this pull request isn't really ready for review.

In fact… it doesn't really look like a good idea IMO. It's more code and doesn't really bring much benefit.